src/resources/js/components/AccountEdit.vue

```diff
@@ -0,0 +1,181 @@
+      :rules="nameRules"
+      label="Prenom"
+    />
+
+    <v-text-field
+      v-model="updateUserInfo.lname"
+      :counter="10"
+      :rules="nameRules"
+      label="Nom"
+    />
+
+    <v-text-field
+      v-model="updateUserInfo.email"
+      :rules="emailRules"
+      label="E-mail"
+    />
+
+    <v-text-field
+      v-model="updateUserInfo.password"
+      :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
+      :type="showPassword ? 'text' : 'password'"
+      label="Mot de passe"
+      value=""
+      class="input-group--focused"
+      @click:append="showPassword = !showPassword"
+    />
+
+    <v-btn
+      :disabled="!valid"
+      color="success"
+      class="mr-4"
+      @click="submitChange"
+    >
+      Appliquer
+    </v-btn>
+
+  </v-form>
+
+  </v-container>
+</template>
+
+<script>
+export default {
+  name: 'AccountEdition',
+  data() {
+    return {
+      valid: true, // if a rules is not fully satisfied this will disable the submit button
+      nameRules: [
+        (v) => v === '' || v.length <= 10 || 'Le nom doit contenir moins de 10 caractères',
+      ],
+      emailRules: [
+        (v) => v === '' || /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,})+$/.test(v) || 'l\'E-mail doit être valide',
+      ],
+      showPassword: false,
+      // object for message management
+      message: {
+        show: false,
+        content: '',
+        type: '',
+      },
+      /*
+       * For now LoggedUser is the current user logged info
+       * this will serve as temporary test and debug value
+       */
+      loggedUser: {},
+      // Object for form field binding
+      updateUserInfo: {
+        fname: '',
+        lname: '',
+        email: '',
+        password: '',
+      },
+    };
+  },
+  beforeMount() {
+    // set Bearer token in header of the future request
+    window.axios.defaults.headers.common = { Authorization: `Bearer ${localStorage.getItem('Authorization-token')}` };
+
+    // get logged user info
+    window.axios
+      .get('/api/v1/me')
+      .then((response) => {
+        this.loggedUser = response.data.data;
+      })
+      .catch((error) => {
+        this.showMessage('error', `La recupération des donnée utilisateur a échoué: ${error}`);
+      });
+  },
+  methods: {
+    submitChange() {
+      // API Url use mirageJS for testing data ,
+      const apiUrl = `/api/user/${this.loggedUser.id}`;
+
+      // real backend URL
+      // const apiUrl = `/api/v1/users/${this.loggedUser.id}`;
+
+      // prepare the data to send
+      const data = this.getDataToSend();
+
+      // Check if the object is Empty
+
+      if (Object.keys(data).length === 0) {
+        this.showMessage('error', 'Vous devez remplir au moins un champs');
+        return;
+      }
+
+      // send http request with axios and catch response or error
+      window.axios.put(apiUrl, data)
+        .then((response) => {
+          // real backend response
+          // this.loggedUser = response.data.data;
```

Comment:
real back-end works, you can un-comment this line

Reply:
ok, it's uncomment now in [b5d02](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/123/commits/b5d0294d19d2a29c924c42a6399d7fac01f14ec4)